src/hipscat/pixel_tree/moc_filter.py

```diff
@@ -0,0 +1,67 @@
+import numba
+import numpy as np
+from mocpy import MOC
+from numba import njit
+
+from hipscat.pixel_tree.pixel_tree import PixelTree
+
+
+def filter_by_moc(
+    tree: PixelTree,
+    moc: MOC,
+) -> PixelTree:
+    """Filters a pixel tree to only include the pixels that overlap with the pixels in the moc
+
+    Args:
+        tree (PixelTree): The tree to perform the filtering on
+        moc (mocpy.MOC): The moc to use to filter
+
+    Returns:
+        A new PixelTree object with only the pixels from the input tree that overlap with the moc.
+    """
+    moc_ranges = moc.to_depth29_ranges
+    tree_29_ranges = tree.tree << (2 * (29 - tree.tree_order))
```

Comment:
Can you add some more comments around this? Anytime there's bit twiddling, I have to stop and think.

Reply:
Sure thing!